include/zim/archive.h

```diff
@@ -541,9 +541,14 @@ namespace zim
    * An `EntryRange` can't be modified and is consequently threadsafe.
    */
   template<EntryOrder order>
-  class LIBZIM_API Archive::iterator : public std::iterator<std::bidirectional_iterator_tag, Entry>
+  class LIBZIM_API Archive::iterator
   {
     public:
+      using iterator_category = std::bidirectional_iterator_tag;
+      using value_type = Entry;
+      using pointer = Entry*;
+      using reference = Entry&;
```

Comment:
> It seems we are in the category of a [InputIterator](https://en.cppreference.com/w/cpp/named_req/InputIterator):
> 
> > LegacyInputIterators only guarantee validity for single pass algorithms: once a LegacyInputIterator i has been incremented, all copies of its previous value may be invalidated.

Yes and no. `LegacyInputIterator` is the lowest capability of iterators. If a type can be claimed to represent an iterator then it automatically satisfies the conceptual guarantees formulated for `LegacyInputIterator` (note, however, that it may also meet the syntax requirements). Thus `Archive::iterator` is a `LegacyInputIterator`

But `Archive::iterator` is far more capable than `LegacyInputIterator`. The problem is that it falls slightly short of complying with the requirements of `LegacyBidirectionalIterator`, which can be easily fixed at the cost of ABI breakage.

Reply:
Discussion continues in later review.
So I mark this one as resolved.